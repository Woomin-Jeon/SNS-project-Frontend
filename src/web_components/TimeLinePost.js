import AddPost from "../components/posts/AddPost";
import ShowPostHome from "../components/posts/ShowPostHome";
import React from "react";

function TimeLinePost({
  postState,
  setPostState,
  currentUserState,
  commentState,
  setCommentState,
  setTopLevelState,
  loginState,
  setLoginState,
}) {
  const { id, friends } = currentUserState;
  const { post: posts } = postState;
  const appropriatePosts = posts.filter(post => post.id === id || friends.includes(post.id));

  return (
    <div className="timeline-post">
      <AddPost
        currentUserState={currentUserState}
        postState={postState}
        setPostState={setPostState}
      />
      <div>
        {appropriatePosts.map((p, index) => (
          <div key={index}>
            <ShowPostHome
              postState={postState}
              setPostState={setPostState}
              currentUserState={currentUserState}
              commentState={commentState}
              setCommentState={setCommentState}
              setTopLevelState={setTopLevelState}
              specificPost={p}
              index={index}
              loginState={loginState}
              setLoginState={setLoginState}
            />
          </div>
        ))}
      </div>
    </div>
  );
}

export default TimeLinePost;
