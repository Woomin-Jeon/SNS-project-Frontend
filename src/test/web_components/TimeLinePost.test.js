import React from "react";
import {shallow} from 'enzyme';
import TimeLinePost from "../../web_components/TimeLinePost";

describe('TimeLinePost', () => {
  let loginState;
  let setLoginState;
  let currentUserState;
  let postState;
  let setPostState;
  let commentState;
  let setCommentState;
  let setTopLevelState;

  beforeEach(() => {
    currentUserState = {
      id: 'Random_ID',
      friends: ['Random_ID'],
    };
    postState = {
      post: [
        {
          id: 'Random_ID'
        }
      ]
    };
  });

  it('renders well', () => {
    const component = shallow(
      <TimeLinePost
        postState={postState}
        setPostState={setPostState}
        currentUserState={currentUserState}
        commentState={commentState}
        setCommentState={setCommentState}
        setTopLevelState={setTopLevelState}
        loginState={loginState}
        setLoginState={setLoginState}
      />
    );

    expect(component).toMatchSnapshot();
  });
});
