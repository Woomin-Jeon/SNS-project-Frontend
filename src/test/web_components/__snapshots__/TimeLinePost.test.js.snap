// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeLinePost renders well 1`] = `ShallowWrapper {}`;
